packages/sui-ssr/server/criticalCss/index.js

```diff
@@ -0,0 +1,71 @@
+import routes from 'routes'
+import {match} from 'react-router'
+import https from 'https'
+import parser from 'ua-parser-js'
+
+const __CACHE__ = {}
+
+const generateMinimalCSSHash = routes => {
+  return routes.reduce((acc, route) => {
+    route.path && (acc += route.path)
+    return acc
+  }, '')
+}
+
+export default withCriticalCSS => (req, res, next) => {
+  if (!withCriticalCSS) {
+    return next()
+  }
+
+  const ua = parser(req.headers['user-agent'])
+  const urlRequest =
+    req.protocol +
+    '://' +
+    (process.env.CRITICAL_CSS_HOST || req.get('host')) +
+    req.url
+
+  const type = ua.device.type
+  const device = type === 'mobile' ? 'm' : type === 'tablet' ? 't' : 'd'
```

Comment:
👏

Reply:
Let's try to avoid having magical strings and a nested ternary in the same line.
 
```
const deviceTypes = {
  desktop: 'd',
  tablet: 't'
  mobile: 'm'
}

const device = deviceType[type] || deviceType.desktop
```

Reply:
Really ?!?!

Reply:
If you're fine with it, let as it is. 👍  I though it could help readability and maintainability of the package.